### PR TITLE
Add .stickler.yml config file

### PR DESCRIPTION
**Pull request by jtraulle**
# New Add SticklerCI

SticklerCI helps enforcing a code style policy the easy way by automatically fixing most code style issues on Pull Requests.

This config file simply use the `dev/setup/codesniffer/ruleset.xml` Dolibarr custom ruleset file and will be used only if SticklerCI GitHub App is enabled after merging the config file :
- https://github.com/apps/stickler-ci

### CODE REVIEW

**Comment by jtraulle** [comment]
You should also disable the Hound GitHub App from https://github.com/Dolibarr/dolibarr/settings/installations by clicking on **Configure** next to the Hound app and clicking **Uninstall** at the bottom of the page as it is currently unused in Dolibarr since there is no `.hound.yml` file in the root of the Dolibarr repo 😉 

------

SticklerCI does basically the same thing as Hound but allow to use a custom ruleset for PhpCodeSniffer rules, works only on Pull Requests and can apply automatic fixes similar to what does `phpcbf` (this is not the case of Hound).

File: .stickler.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+linters:
+  phpcs:
+      standard: 'dev/setup/codesniffer/ruleset.xml'
```
Comment (jtraulle) [inline .stickler.yml]: I am currently validating that this is the right way to define a custom ruleset. I have [twitted](https://twitter.com/jtraulle/status/1166072293497483264) to @sticklerci and I am currently waiting for a reply.

Maybe we need to add a symbolic link `phpcs.xml` at the root of the project that direct to `dev/setup/codesniffer/ruleset.xml`. [Doc](https://stickler-ci.com/docs#php) is unclear to me, that's why I asked for confirmation [on Twitter](https://twitter.com/jtraulle/status/1166072293497483264).

Comment (jtraulle) [reply .stickler.yml]: [It is OK](https://twitter.com/sticklerci/status/1166138847035613184) :slightly_smiling_face:

**Comment by jtraulle** [comment]
@eldy This in now ready to be merged :slightly_smiling_face: 

To further increase the speed of the Travis-CI builds for Pull Requests, could you also enable the **Auto cancel pull request builds** [in the settings of Travis-CI](https://travis-ci.org/dolibarr/dolibarr/settings) to only build the last pushed commit on the Pull Request branch and skip previously pushed commits that are waiting to run (this seems to be already enabled for branches but not for pull requests). Thanks.

See https://docs.travis-ci.com/user/customizing-the-build/#building-only-the-latest-commit for more info.

**Comment by eldy** [comment]
@jtraulle 
If i enable the autocancel on travis for Pull Requests, if 2 different people make 2 PR with only few seconds of difference. I think this means the first one will not run travis. So i and the first people won't have the result of travis ci only on the first PR. Am i wrong ?
Or the cancel is only for "same author" or same "PR" ?

**Comment by jtraulle** [comment]
@eldy Yes, this is working as a per branch basis so the cancellation only happen for previous commits on the **same** Pull Request :wink:

**Comment by eldy** [comment]
So it is enabled now

**Comment by jtraulle** [comment]
Thanks :slightly_smiling_face:

**Comment by jtraulle** [comment]
@eldy When enabling the SticklerCI GitHub App, did you authorize properly the write access to code ?

I have saw earlier that SticklerCI returned a **Unable to apply fixers. Could not push fix commit because permission was denied** error to @aspangaro PR. Could you check ?

Except for the auto-fix feature, it works well, that's great :slightly_smiling_face: 

Example : https://github.com/Dolibarr/dolibarr/pull/11767/checks?sha=958361b45ea49d29833864ad87e42333ff538379

**Pull request merged**